### PR TITLE
Database support

### PR DESCRIPTION
**Pull request by vifino**
This PR adds database support:

`sql.drivers()` returns the available drivers in table form, should return `{"mysql", "postgres", "sqlite3"}`.

`sql.open(drivername, source)` creates a new DB wrapper. A sqlite3 instance in memory would be as simple as the following Example:

``` lua
local db = sql.open("sqlite3", ":memory:")
```

The DB wrapper has a few fields:
- `con`
  - Please don't touch. This is the internal connection object, iterating it doesn't work, calling things wrong crashes carbon, etc..
- `close`
  - Closes automatically opened connections, however, this should be barely useful, since db wrappers are supposed to be long lasting and used by multiple threads.
- `exec`
  - Executes statements:
    - `db:exec("CREATE TABLE users (name text, age int)")`
  - Can also take extra arguments and replace `$1`, `$2`, etc.. respectively.
- `query`
  - Probably the most useful.
  - Also takes format arguments similar to `exec`.
  - Example query:

``` lua
local rows = assert(db:query("SELECT * FROM users"))
for rowno=1, rows.n do
  print("Showing row no. "..tostring(rowno))
  for k, v in pairs(rows[rowno]) do
    print("> "..k, v)
  end
end
```
- `prepare`
  - Pretty much `query`, but you prepare a statement beforehand, then execute it repeatedly.

``` lua
local tables = {"users", "administrators"}
local statement = db:prepare("SELECT * FROM $1")
for _, tbl in pairs(tables) do
  local rows = statement:query(tbl)
  for rowno=1, rows.n do
    print(tbl..": Showing row no. "..tostring(rowno))
    for k, v in pairs(rows[rowno]) do
      print("> "..k, v)
    end
  end
end
```

See also: `tests/sqltest.lua`

---

@bauen1 made a wiki entry for this PR: https://github.com/carbonsrv/carbon/wiki/API:-sql

### CODE REVIEW

**Comment by vifino** [comment]
I suppose this is good to go.

As far as I can tell, @bauen1 tested this too (apart from me, duh.).

What do you think, @bauen1?

**Comment by bauen1** [comment]
Works for me

**Comment by vifino** [comment]
Awesome.

File: builtin/libs/wrappers/sql.lua [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+			end
+
+			return convert_rows(rows)
+		end,
+		prepare = function(origself, stmt)
+			if not stmt then
+				error("SQL: prepare needs statement!", 0)
+			end
+
+			do_ping(self.con)
```
Comment (bauen1) [inline builtin/libs/wrappers/sql.lua]: Right here

Comment (vifino) [inline builtin/libs/wrappers/sql.lua]: Damn. I'll fix it ASAP.

Comment (vifino) [inline builtin/libs/wrappers/sql.lua]: Fixed in master.

**Pull request merged**